pkg/trait/deployment.go

```diff
@@ -418,7 +419,8 @@ func (t *deploymentTrait) getDeploymentFor(e *Environment) *appsv1.Deployment {
 	//
 
 	VisitConfigurations("configmap", e.Context, e.Integration, func(cmName string) {
-		refName := "integration-cm-" + strings.ToLower(cmName)
+		refName := xid.New().String()
```

Comment:
We don't reconcile when subresources change, but if we start doing it in the future, this creates problems as it's random. Wdyt @lburgazzoli ?

Reply:
done